src/include/miopen/any_solver.hpp

```diff
@@ -243,51 +246,59 @@ struct AnySolver
+            return miopen::solver::FindSolution(value, ctx, db, invoke_ctx, perf_cfg);
         };
 
-        std::string GetPerfCfgParams(const ConvolutionContext& ctx, Db& db, std::true_type) const
+        std::optional<std::string>
+        GetPerfCfgParams(const ConvolutionContext& ctx, PerformanceDb& db, std::true_type) const
         {
             using PerformanceConfig = decltype(value.GetDefaultPerformanceConfig(ctx));
             PerformanceConfig config{};
             if(db.Load(ctx.problem, value.SolverDbId(), config))
             {
-                MIOPEN_LOG_I2("Perf Db: Record Loaded: " << value.SolverDbId());
+                MIOPEN_LOG_I2("Perf PerformanceDb: Record Loaded: " << value.SolverDbId());
                 if(value.IsValidPerformanceConfig(ctx, config))
                 {
                     return config.ToString();
                 }
-                MIOPEN_LOG_I2("Perf Db: Invalid Config: " << value.SolverDbId());
+                MIOPEN_LOG_I2("Perf PerformanceDb: Invalid Config: " << value.SolverDbId());
             }
             else if(!value.AltSolverDbId().empty() &&
                     db.Load(ctx.problem, value.AltSolverDbId(), config))
             {
-                MIOPEN_LOG_I("Perf Db: alternate record loaded: " << value.AltSolverDbId());
+                MIOPEN_LOG_I(
+                    "Perf PerformanceDb: alternate record loaded: " << value.AltSolverDbId());
                 if(value.IsValidPerformanceConfig(ctx, config))
                 {
                     return config.ToString();
                 }
-                MIOPEN_LOG_I2("Perf Db: Invalid alternate record from Perf Db: "
-                              << value.AltSolverDbId() << ": " << config);
+                MIOPEN_LOG_I2(
+                    "Perf PerformanceDb: Invalid alternate record from Perf PerformanceDb: "
+                    << value.AltSolverDbId() << ": " << config);
             }
 
-            MIOPEN_LOG_I2("Perf Db: Failed Loading, Using Default: " << value.SolverDbId());
+            MIOPEN_LOG_I2(
+                "Perf PerformanceDb: Failed Loading, Using Default: " << value.SolverDbId());
             config = value.GetDefaultPerformanceConfig(ctx);
             return config.ToString();
         }
-        std::string
-        GetPerfCfgParams(const ConvolutionContext& ctx, const Db& db, std::false_type) const
+
+        std::optional<std::string> GetPerfCfgParams(const ConvolutionContext& ctx,
+                                                    const PerformanceDb& db,
+                                                    std::false_type) const
         {
-            MIOPEN_LOG_I2("Perf Db: No Config: " << value.SolverDbId());
+            MIOPEN_LOG_I2("Perf PerformanceDb: No Config: " << value.SolverDbId());
             std::ignore = ctx;
             std::ignore = db;
-            return "";
+            return std::nullopt;
```

Comment:
This change necessitates the multiline changes to conv_fin.hpp. Is there a reason to return null here? I would suggest reverting this line so the Fin changes aren't necessary.

Reply:
If we use an empty string as no-value marker, we need to guarantee that there would be no solver which may have an empty string as a valid perf config value, but it is not default. This may lead to weird bugs which would be difficult to debug. I thought, I have outlined it somewhere before and it just got overlooked, but I can't find that.
But you are probably right. At the moment that would require a lot of changes to be meaningful. I should change it to `""` with no optional.